### PR TITLE
[MERGE] mail, *: several UX/UI improvements

### PR DESCRIPTION
**Pull request by alexkuhn**
*: bus, im_livechat, mail_bot, website_livechat

Several improvements for Discuss app:

1. Improved "is typing..." notification
2. Merged private & public channels
3. Quick channel search
4. No push notification right after accepting them, and transparent odoo bot icon in native notification
5. Selected emojis are inserted in composer based on cursor position
6. Set surname to people for DMs name
7. Discard replying to a message in Inbox
8. Broadcast "seen" notification ('v' and 'vv' on messages)

[1] [image1](https://drive.google.com/file/d/11_H6ZJh0wFaegAV0fTLdc68z37nhVKnW/view?usp=sharing), [image2](https://drive.google.com/file/d/1s9g_hUCMesuTLAqpeGMwLWzHCYnzanr6/view?usp=sharing)
[2] [image1](https://drive.google.com/file/d/1WUBqc2lScfWV8Yrtq6Zk4MO4jC0CdaiG/view?usp=sharing), [image2](https://drive.google.com/file/d/13XVcW0Nookwea_4xz02mIwPi5dPhXu18/view?usp=sharing)
[3] [image1](https://drive.google.com/file/d/1FuV3xH9cK6qR7ljd5vELOHc4icvHIJTF/view?usp=sharing), [image2](https://drive.google.com/file/d/1hwuu5ejlj9ou02NF5XmL-XHTiD-Ty6XT/view?usp=sharing)
[4] [image](https://drive.google.com/file/d/1si1TYtCpdyOgw9IRaMWmxuUb1d8M4ccF/view?usp=sharing)
[5] [video](https://drive.google.com/file/d/19knQ47F1wzlNvxT2nJH3A9V34wO1D9Bn/view?usp=sharing)
[6] [image1](https://drive.google.com/file/d/11ot_DtySN2cIN_bM8n7fhFV2JXNoz0lH/view?usp=sharing), [image2](https://drive.google.com/file/d/1aNyapzKePIRg903woCo3kXJhy-ZFwfEG/view?usp=sharing), [image3](https://drive.google.com/file/d/1AtVlOG67YjsS8ax1ErWhXfSDOtKhLQhC/view?usp=sharing)
[7] [image](https://drive.google.com/file/d/1U-VEut7M7-t9WzdKVagfyy6ZrbOJrflp/view?usp=sharing)
[8] [image1](https://drive.google.com/file/d/1_32q8iqPBhW4sPhAtY5mIO968gj1o9o-/view?usp=sharing), [image2](https://drive.google.com/file/d/1Ne6HMEgJxHAWYN3e3SlrF6Qqg70LDvAW/view?usp=sharing)

Task-ID 1881001

### CODE REVIEW

File: addons/mail/models/mail_channel.py [hunk 1]
```diff
@@ -537,10 +539,10 @@ def channel_info(self, extra_info = False):
             :rtype : list(dict)
         """
         channel_infos = []
-        partner_channels = self.env['mail.channel.partner']
+        my_partner_channels = self.env['mail.channel.partner']
```
Comment (Xavier-Do) [inline addons/mail/models/mail_channel.py]: all_partner_channels

File: addons/mail/static/src/js/composers/basic_composer.js [hunk 2]
```diff
@@ -663,15 +675,26 @@ var BasicComposer = Widget.extend({
...
-        this.$input.val(this.$input.val() + " " + $(ev.currentTarget).data('emoji') + " ");
+        var cursorPosition = this.getSelectionPositions();
+        var inputVal = this.$input.val();
+        var leftSubstring = inputVal.substring(0, cursorPosition.start);
+        var rightSubstring = inputVal.substring(cursorPosition.end);
+        var emoji = $(ev.currentTarget).data('emoji');
+        var newInputVal = leftSubstring + " " + emoji + " ";
+        var newCursorPosition = newInputVal.length;
+        newInputVal += rightSubstring;
+        this.$input.val(leftSubstring + " " + emoji + " " + rightSubstring);
```
Comment (Xavier-Do) [inline addons/mail/static/src/js/composers/basic_composer.js]: newInputVal 

or

```js
var newInputVal  = [leftSubstring , $(ev.currentTarget).data('emoji'), rightSubstring].join(' ');
var newCursorPosition = newInputVal.length - rightSubstring.length;
```

File: addons/mail/static/src/js/composers/extended_composer.js [hunk 3]
```diff
@@ -67,6 +74,33 @@ var ExtendedComposer = BasicComposer.extend({
...
         this.$('.o_composer_subject input').val(subject);
     },
+    /**
+     * Show the button 'discard' next to the send button. This is useful while
+     * replying to a message, when in the end we do not want to reply.
+     *
+     * This is ignored if discard button is already there.
+     */
+    showDiscard: function () {
+        if (this.$('.o_composer_button_discard').length) {
```
Comment (Xavier-Do) [inline addons/mail/static/src/js/composers/extended_composer.js]: \> 0

Comment (Xavier-Do) [reply addons/mail/static/src/js/composers/extended_composer.js]: why not exist by default but hidden?

File: addons/mail/static/src/js/composers/extended_composer.js [hunk 4]
```diff
@@ -67,6 +74,33 @@ var ExtendedComposer = BasicComposer.extend({
...
+    },
+    /**
+     * Hide the button 'discard' next to the send button. This is useful when
+     * we were replying to a message from a certain thread, then we discard it
+     * with some other actions (i.e. switch thread). The discard button only
+     * makes sense during the selection of a message.
+     *
+     * This is ignored if discard button is not there.
+     */
+    hideDiscard: function () {
```
Comment (Xavier-Do) [inline addons/mail/static/src/js/composers/extended_composer.js]: hide instead of remove

File: addons/mail/static/src/js/discuss.js [hunk 5]
```diff
@@ -111,6 +111,72 @@ var PartnerInviteDialog = Dialog.extend({
...
+            }],
+        });
+    },
+    /**
+     * @override
+     * @returns {$.Promise}
+     */
+    start: function () {
+        var channel = this.call('mail_service', 'getChannel', this._channelID);
+        this.$('input').val(channel.getName());
```
Comment (Xavier-Do) [inline addons/mail/static/src/js/discuss.js]: escape

File: addons/mail/static/src/js/services/mail_window_manager.js [hunk 6]
```diff
@@ -668,7 +668,7 @@ MailManager.include({
         if (thread.isChannel()) {
             // call getMentionpartnerSuggestions in order to correctly fetch members
             thread.getMentionPartnerSuggestions().then(function () {
-                threadWindow.renderTypingNotificationBar();
+                // todo: show is typing icon in title
```
Comment (Xavier-Do) [inline addons/mail/static/src/js/services/mail_window_manager.js]: todo?

**Comment by Xavier-Do** [comment]
robodoo r+

**Comment by Xavier-Do** [comment]
Robodoo rebase-merge

**Comment by alexkuhn** [comment]
robodoo rebase-merge

**Comment by alexkuhn** [comment]
robodoo r+

**Comment by alexkuhn** [comment]
damn... @Xavier-Do Could you please ask rebase-merge again, no capital letter for our humble butler? Thanks!

**Comment by Xavier-Do** [comment]
robodoo rebase-merge

**Comment by alexkuhn** [comment]
robodoo retry

**Comment by alexkuhn** [comment]
robodoo retry

**Pull request closed without merging**